base/MAPL_SwathGridFactory.F90

```diff
@@ -0,0 +1,1259 @@
+
+      character(:), allocatable :: lon_name
+      character(:), allocatable :: lat_name
+      character(:), allocatable :: lev_name
+      integer :: i
+      logical :: hasLon, hasLat, hasLongitude, hasLatitude, hasLev,hasLevel,regLat,regLon
+      real(kind=REAL64) :: del12,delij
+
+      integer :: i_min, i_max
+      real(kind=REAL64) :: d_lat, d_lat_temp, extrap_lat
+      logical :: is_valid, use_file_coords, compute_lons, compute_lats
+
+      write(6,*) 'nail inside    subroutine initialize_from_file_metadata'
+      _UNUSED_DUMMY(unusable)
+
+      if (present(force_file_coordinates)) then
+         use_file_coords = force_file_Coordinates
+      else
+         use_file_coords = .false.
+      end if
+
+      ! Cannot assume that lats and lons are evenly spaced
+
+      associate (im => this%im_world, jm => this%jm_world, lm => this%lm)
+         lon_name = 'lon'
+         hasLon = file_metadata%has_dimension(lon_name)
+         if (hasLon) then
+            im = file_metadata%get_dimension(lon_name, rc=status)
+            _VERIFY(status)
+         else
+            lon_name = 'longitude'
+            hasLongitude = file_metadata%has_dimension(lon_name)
+            if (hasLongitude) then
+               im = file_metadata%get_dimension(lon_name, rc=status)
+               _VERIFY(status)
+            else
+               _FAIL('no longitude coordinate')
+            end if
+         end if
+         lat_name = 'lat'
+         hasLat = file_metadata%has_dimension(lat_name)
+         if (hasLat) then
+            jm = file_metadata%get_dimension(lat_name, rc=status)
+            _VERIFY(status)
+         else
+            lat_name = 'latitude'
+            hasLatitude = file_metadata%has_dimension(lat_name)
+            if (hasLatitude) then
+               jm = file_metadata%get_dimension(lat_name, rc=status)
+               _VERIFY(status)
+            else
+               _FAIL('no latitude coordinate')
+            end if
+         end if
+         hasLev=.false.
+         hasLevel=.false.
+         lev_name = 'lev'
+         hasLev = file_metadata%has_dimension(lev_name)
+         if (hasLev) then
+            lm = file_metadata%get_dimension(lev_name,rc=status)
+            _VERIFY(status)
+         else
+            lev_name = 'levels'
+            hasLevel = file_metadata%has_dimension(lev_name)
+            if (hasLevel) then
+               lm = file_metadata%get_dimension(lev_name,rc=status)
+               _VERIFY(status)
+            end if
+         end if
+
+         ! TODO: check radians vs degrees.  Assume degrees for now.
+
+!        28-Feb-2023 disable array retrieval from metadata in Swath
+!         
+!        ! TODO: modify CoordinateVariable so that get_coordinate_data() is overloaded
+!        ! for different types (as subroutine) to avoid casting here.
+!        ! TODO: add get_coordinate_variable() interface to avoid the need to cast
+!        v => file_metadata%get_coordinate_variable(lon_name, rc=status)
```

Comment:
Especially delete long sections of commented out code.  Or is this code that is just not debugged yet?

Reply:
It is not debugged. @bena-nasa and I were discussing if `sub initialize_from_file_metadata`  needs to import lat/lon data from metadata into the SwathGridFactory class.  They were not read-in within the Tripolar case.